discovery-provider/src/api_helpers.py

```diff
@@ -1,17 +1,55 @@
-from flask import jsonify
+import logging
+import json
+import datetime
 import redis
+from web3 import Web3
+from web3.auto import w3
+from eth_account.messages import encode_defunct
+from flask import jsonify
+
 from src.utils.config import shared_config
 from src.utils.redis_constants import latest_block_redis_key, most_recent_indexed_block_redis_key
 
 redis_url = shared_config["redis"]["url"]
 redis = redis.Redis.from_url(url=redis_url)
 
+logger = logging.getLogger(__name__)
+METADATA_FIELDS = ['success', 'latest_indexed_block', 'latest_chain_block']
+API_SIGNING_FIELDS = ['timestamp', 'signature']
+
+# Subclass JSONEncoder
+class DateTimeEncoder(json.JSONEncoder):
+    # Override the default method
+    def default(self, o): # pylint: disable=E0202
+        if isinstance(o, (datetime.date, datetime.datetime)):
+            # TODO - the Z is required in JS date format
+            return o.isoformat() + " Z"
+        return json.JSONEncoder.default(self, o)
 
 def error_response(error, error_code=500):
     return jsonify({'success': False, 'error': error}), error_code
 
-
 def success_response(response_entity=None, status=200):
+    response_dictionary = response_dict_with_metadata(response_entity)
+    return jsonify(response_dictionary), status
+
+# Create a response dict with just data, signature, and timestamp
+def response_with_signature(response_entity=None, status=200):
+    # Make copy of response entity
+    response_dictionary = {**response_entity}
+    generate_and_set_sig_and_timestamp(response_dictionary, response_entity)
+    return jsonify(response_dictionary), status
+
+# Create a response dict with metadata, data, signature, and timestamp
+def success_response_with_signature(response_entity=None, status=200):
```

Comment:
i also think we should probably include the owner_wallet and version as part of every response in the metadata object

Reply:
sure! im guessing we won't deprecate `/version` route?

Reply:
eventually i think we should deprecate it.  the reason this is useful is because sp's who want to verify can just run a single route and check wallet, version and signature all in one go with no external dependencies

Reply:
do we want to add owner_wallet and version to the routes in health_check.py?